src/DI/Compiler.php

```diff
@@ -106,13 +111,15 @@ public function processParameters()
 	/** @internal */
 	public function processExtensions()
 	{
-		for ($i = 0; $slice = array_slice($this->extensions, $i, 1, TRUE); $i++) {
-			$name = key($slice);
+		foreach ($this->extensions as $name => $extension) {
 			if (isset($this->config[$name])) {
 				$this->config[$name] = $tmp = Helpers::expand($this->config[$name], $this->builder->parameters);
 				unset($tmp['services']);
 				$this->extensions[$name]->setConfig($tmp);
 			}
+		}
+		for ($i = 0; $slice = array_slice($this->extensions, $i, 1, TRUE); $i++) {
+			$name = key($slice);
```

Comment:
Yes, this should call setConfig in loop in processParameters() instead of addExtension.